examples/cloudsuite/kit/cloudsuite.py

```diff
@@ -0,0 +1,475 @@
+    """
+
+    def __init__(
+        self,
+        src_dir: PathType,
+        command_wrappers: Iterable[CommandWrapper],
+        command_attachments: Iterable[CommandAttachment],
+        shared_libs: Iterable[SharedLib],
+        pre_run_hooks: Iterable[PreRunHook],
+        post_run_hooks: Iterable[PostRunHook],
+        platform: Platform | None = None,
+        server_platform: Platform | None = None,
+        web_server_platform: Platform | None = None,
+    ) -> None:
+        super().__init__(
+            command_wrappers=command_wrappers,
+            command_attachments=command_attachments,
+            shared_libs=shared_libs,
+            pre_run_hooks=pre_run_hooks,
+            post_run_hooks=post_run_hooks,
+        )
+
+        if platform is not None:
+            self.platform = platform  # TODO Warning! overriding upper class platform
+        self.server_platform = server_platform
+        self.web_server_platform = web_server_platform
+
+        bench_src_path = pathlib.Path(src_dir)
+        if not self.platform.comm.isdir(bench_src_path) and self.platform.comm.isfile(
+            bench_src_path / "benchmarks/web-serving/db_server/Dockerfile"
+        ):
+            raise ValueError(
+                f"Invalid Cloudsuite source path: {bench_src_path}\n"
+                "src_dir argument can be defined manually."
+            )
+        self._bench_src_path = bench_src_path
+
+    @property
+    def bench_src_path(self) -> pathlib.Path:
+        return self._bench_src_path
+
+    @staticmethod
+    def get_build_var_names() -> List[str]:
+        return [
+            "generator_seed",
+            "nb_threads",
+        ]
+
+    @staticmethod
+    def get_run_var_names() -> List[str]:
+        return [
+            "nb_threads",
+        ]
+
+    @staticmethod
+    def get_tilt_var_names() -> List[str]:
+        return []
+
+    def dependencies(self) -> List[PackageDependency]:
+        return super().dependencies() + [
+            PackageDependency("build-essential"),
+            PackageDependency("libhwloc-dev"),
+        ]
+
+    def build_tilt(self, **kwargs) -> None:
+        raise NotImplementedError("Tilt is not necessary for this benchmark.")
+
+    def prebuild_bench(
+        self,
+        **_kwargs,
+    ) -> None:
+        pass
+
+    def build_bench(  # pylint: disable=arguments-differ
+        self,
+        nb_threads: int = 2,
+        generator_seed: int = 0,
+        **_kwargs,
+    ) -> None:
+
+        random.seed(generator_seed)
+        seeds = [random.randrange(2**20 - 1) for j in range(0, nb_threads)]
+
+        seeds_str = ""
+
+        for i, s in enumerate(seeds):
+            seeds_str = seeds_str + f"SEED{i}={s}\\\\n"
+
+        self.platform.comm.shell(
+            command=f"printf {seeds_str} > _tmp_benchkit_seeds.txt",
+            current_dir="~/",
+        )
+
+        # 1) Move fabandriver.jar and Web20Driver.java.in into the right places
```

Comment:
put these comments at the right place where these operations are actually executed

Reply:
I've moved this and also expanded a bit on each step taken. Could you check if the current version is better?